src/policy/mallocspace/metadata.rs

```diff
@@ -1,61 +1,137 @@
 use crate::util::metadata::load_metadata;
 use crate::util::metadata::side_metadata;
+#[cfg(target_pointer_width = "64")]
+use crate::util::metadata::side_metadata::metadata_address_range_size;
+#[cfg(target_pointer_width = "32")]
+use crate::util::metadata::side_metadata::metadata_bytes_per_chunk;
 use crate::util::metadata::side_metadata::SideMetadataContext;
 use crate::util::metadata::side_metadata::SideMetadataSpec;
-#[cfg(target_pointer_width = "64")]
+use crate::util::metadata::side_metadata::GLOBAL_SIDE_METADATA_BASE_ADDRESS;
 use crate::util::metadata::side_metadata::LOCAL_SIDE_METADATA_BASE_ADDRESS;
 use crate::util::metadata::store_metadata;
+use crate::util::metadata::MetadataSpec;
 use crate::util::Address;
 use crate::util::ObjectReference;
+use crate::util::{constants, conversions, metadata};
 use crate::vm::{ObjectModel, VMBinding};
-
-use std::collections::HashSet;
-use std::sync::RwLock;
+use std::sync::atomic::Ordering;
+use std::sync::Mutex;
 
 lazy_static! {
-    pub static ref ACTIVE_CHUNKS: RwLock<HashSet<Address>> = RwLock::default();
+    pub(super) static ref CHUNK_METADATA: SideMetadataContext = SideMetadataContext {
+        global: metadata::extract_side_metadata(&[MetadataSpec::OnSide(
+            ACTIVE_CHUNK_METADATA_SPEC
+        )]),
+        local: vec![],
+    };
+    static ref CHUNK_MAP_LOCK: Mutex<()> = Mutex::new(());
 }
 
+/// Metadata spec for the active chunk byte
+///
+/// This metadata is mapped eagerly (as opposed to lazily like the others),
+/// hence a separate `SideMetadata` instance is required.
+///
+/// This is a global side metadata spec even though it is used only by MallocSpace as
+/// we require its space to be contiguous and mapped only once. Otherwise we risk
+/// overwriting the previous mapping.
+pub(crate) const ACTIVE_CHUNK_METADATA_SPEC: SideMetadataSpec = SideMetadataSpec {
+    is_global: true,
+    offset: GLOBAL_SIDE_METADATA_BASE_ADDRESS.as_usize(),
+    log_num_of_bits: 3,
+    log_min_obj_size: LOG_BYTES_IN_CHUNK as usize,
+};
+
 /// This is the metadata spec for the alloc-bit.
 ///
-/// An alloc-bit is required per min-object-size aligned address , rather than per object, and can only exist as side metadata.
+/// An alloc-bit is required per min-object-size aligned address, rather than per object, and can only exist as side metadata.
 ///
 /// The other metadata used by MallocSpace is mark-bit, which is per-object and can be kept in object header if the VM allows it.
 /// Thus, mark-bit is vm-dependant and is part of each VM's ObjectModel.
 ///
-#[cfg(target_pointer_width = "32")]
 pub(crate) const ALLOC_SIDE_METADATA_SPEC: SideMetadataSpec = SideMetadataSpec {
     is_global: false,
-    offset: 0,
+    offset: if cfg!(target_pointer_width = "64") {
+        LOCAL_SIDE_METADATA_BASE_ADDRESS.as_usize()
+    } else {
+        0
+    },
     log_num_of_bits: 0,
     log_min_obj_size: constants::LOG_MIN_OBJECT_SIZE as usize,
 };
 
+/// Metadata spec for the active page byte
```

Comment:
Again, probably a short explanation of what `active page byte` is, is useful here